presto-hive/src/main/java/io/prestosql/plugin/hive/UnregisterPartitionProcedure.java

```diff
@@ -0,0 +1,136 @@
+
+public class UnregisterPartitionProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle UNREGISTER_PARTITION = methodHandle(
+            UnregisterPartitionProcedure.class,
+            "unregisterPartition",
+            ConnectorSession.class,
+            String.class,
+            String.class,
+            List.class,
+            List.class);
+
+    private final Supplier<TransactionalMetadata> hiveMetadataFactory;
+    private final HiveMetastoreClosure metastore;
+
+    @Inject
+    public UnregisterPartitionProcedure(Supplier<TransactionalMetadata> hiveMetadataFactory, HiveMetastore metastore)
+    {
+        this.hiveMetadataFactory = requireNonNull(hiveMetadataFactory, "hiveMetadataFactory is null");
+        this.metastore = new HiveMetastoreClosure(requireNonNull(metastore, "metastore is null"));
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "unregister_partition",
+                ImmutableList.of(
+                        new Procedure.Argument("schema_name", VARCHAR),
+                        new Procedure.Argument("table_name", VARCHAR),
+                        new Procedure.Argument("partition_columns", new ArrayType(VARCHAR)),
+                        new Procedure.Argument("partition_values", new ArrayType(VARCHAR))),
+                UNREGISTER_PARTITION.bindTo(this));
+    }
+
+    public void unregisterPartition(ConnectorSession session, String schemaName, String tableName, List<Object> partitionColumnNames, List<Object> partitionValues)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doUnregisterPartition(session, schemaName, tableName, partitionColumnNames, partitionValues);
+        }
+    }
+
+    private void doUnregisterPartition(ConnectorSession session, String schemaName, String tableName, List<Object> partitionColumnNames, List<Object> partitionValues)
+    {
+        HiveIdentity identity = new HiveIdentity(session);
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+
+        Table table = metastore.getTable(identity, schemaName, tableName)
+                .orElseThrow(() -> new TableNotFoundException(schemaTableName));
+
+        if (table.getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+            throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, "Table is a view: " + schemaTableName);
+        }
+
+        if (table.getPartitionColumns().isEmpty()) {
+            throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, "Table is not partitioned: " + schemaTableName);
+        }
+
+        List<String> actualPartitionColumnNames = table.getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+
+        if (!Objects.equals(partitionColumnNames, actualPartitionColumnNames)) {
```

Comment:
Case insensitive?

Reply:
I'll make a follow-up PR changing this in every procedure (create empty partition as well)